### PR TITLE
Add `textFontWeight` modifier

### LINKED ISSUE #578: Add `.textFontWeight` modifier
In order to support `Tabs` (but not only for that), we should introduce a modifier to mimic the behavior of the native [fontWeight](https://developer.apple.com/documentation/swiftui/view/fontweight(_:)), which is only available from iOS 16.

### PR DESCRIPTION
**Pull request by sjavora**
Closes #578 , needed for #577.

### CODE REVIEW

File: Sources/Orbit/Support/Environment Keys/TextFontWeightKey.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    /// Set the font weight for any text based Orbit component within the view hierarchy.
+    ///
+    /// - Parameters:
+    ///   - fontWeight: A font weight that will be used in text based Orbit components.
+    ///                 Pass `nil` to ignore environment font weight and to allow the system
+    ///                 or the container to provide its own font weight.
+    ///                 If a container-specific override doesn’t exist, the `regular` will be used.
+    func textFontWeight(_ fontWeight: Font.Weight?) -> some View {
+        environment(\.textFontWeight, fontWeight)
```
Comment (PavelHolec) [inline Sources/Orbit/Support/Environment Keys/TextFontWeightKey.swift:26]: `@available obsoleted iOS16, renamed: "Use native fontWeight()"` could be added to notify about this being only a workaround for lower iOS versions.

**Review by PavelHolec — APPROVED** [review]

File: Sources/Orbit/Components/Text.swift [hunk 2]
```diff
@@ -106,8 +107,7 @@ public struct Text: View, FormattedTextBuildable {
         if #available(iOS 16.0, *), let isBold {
             return text
                 .bold(isBold)
-        }
-        else if let isBold, isBold {
+        } else if let isBold, isBold {
```
Comment (username0x0a) [inline Sources/Orbit/Components/Text.swift:110]: `} else if isBold == true {` directly? 🤔

Also, aren't we piling up unnecessary attributes (`Text(...).bold(false).italic(false)`) or it is built up for a reason to maybe override it in the underlying hierarchy? Not sure whether this is applicable on some concatenation atm.

Comment (sjavora) [reply Sources/Orbit/Components/Text.swift:110]: I'm going with "not my problem" on this one... I just saw suboptimal formatting 😅

Comment (username0x0a) [reply Sources/Orbit/Components/Text.swift:110]: Legit I guess 😄👌

**Pull request merged**